krm-functions/lib/condkptsdk/inventory_diff_test.go

```diff
@@ -0,0 +1,543 @@
+		// set own new resource
+		inv.set(&gvkKindCtx{gvkKind: ownGVKKind}, []corev1.ObjectReference{
+			{APIVersion: "a", Kind: "a", Name: "a"},
+			{APIVersion: tt.input.apiVersion, Kind: tt.input.kind, Name: tt.input.name},
+		}, kubeObjectMade, true)
+	}
+	diffList, _ := inv.diff()
+	if len(diffList) == 0 {
+		t.Error("expected a diff")
+	}
+	for _, diff := range diffList {
+		if len(diff.createObjs) != 2 {
+			t.Errorf("TestDiffWithSpecToDelete: -want 2, +got:\n%v", len(diff.createObjs))
+		}
+		if len(diff.deleteObjs) != 0 {
+			t.Errorf("TestDiffWithSpecToDelete: -want 0, +got:\n%v", len(diff.deleteObjs))
+		}
+		if len(diff.updateObjs) != 0 {
+			t.Errorf("TestDiffWithSpecToUpdate: -want 0, +got:\n%v", len(diff.updateObjs))
+		}
+	}
+}
+
+func TestDiffWithSpecToDelete(t *testing.T) {
+	inv, err := newInventory(&Config{
+		For:                corev1.ObjectReference{APIVersion: "a", Kind: "a"},
+		GenerateResourceFn: GenerateResourceFnNop,
+	})
+	if err != nil {
+		assert.NoError(t, err)
+	}
+	type object struct {
+		apiVersion string
+		kind       string
+		name       string
+		dummy      string
+		spec       string
+	}
+
+	test1 := []struct {
+		input object
+	}{
+		{
+			input: object{
+				apiVersion: "a",
+				kind:       "b",
+				name:       "c",
+				dummy:      "sample",
+				spec:       "test1",
+			},
+		},
+		{
+			input: object{
+				apiVersion: "d",
+				kind:       "e",
+				name:       "f",
+				spec:       "test2",
+				dummy:      "sample",
+			},
+		},
+	}
+	for _, tt := range test1 {
+		ipa := &nadv1.NetworkAttachmentDefinition{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: tt.input.apiVersion,
+				Kind:       tt.input.kind,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: tt.input.name,
+			},
+			Spec: nadv1.NetworkAttachmentDefinitionSpec{
+				Config: tt.input.spec,
+			},
+		}
+		byteStream, _ := yaml.Marshal(ipa)
+		kubeObjectMade, _ := fn.ParseKubeObject(byteStream)
+		// set for exisiting resource
+		inv.set(&gvkKindCtx{gvkKind: forGVKKind}, []corev1.ObjectReference{
+			{APIVersion: "a", Kind: "a", Name: "a"},
+		}, kubeObjectMade, false)
+		byteStream, _ = yaml.Marshal(ipa)
+		kubeObjectMade, _ = fn.ParseKubeObject(byteStream)
+		// set own new resource
+		inv.set(&gvkKindCtx{gvkKind: ownGVKKind}, []corev1.ObjectReference{
+			{APIVersion: "a", Kind: "a", Name: "a"},
+			{APIVersion: tt.input.apiVersion, Kind: tt.input.kind, Name: tt.input.name},
+		}, kubeObjectMade, false)
+	}
+	diffList, _ := inv.diff()
+	if len(diffList) == 0 {
+		t.Error("expected a diff")
+	}
+	for _, diff := range diffList {
+		if len(diff.createObjs) != 0 {
+			t.Errorf("TestDiffWithSpecToDelete: -want 0, +got:\n%v", len(diff.createObjs))
+		}
+		if len(diff.deleteObjs) != 2 {
+			t.Errorf("TestDiffWithSpecToDelete: -want 2, +got:\n%v", len(diff.deleteObjs))
+		}
+		if len(diff.updateObjs) != 0 {
+			t.Errorf("TestDiffWithSpecToUpdate: -want 0, +got:\n%v", len(diff.updateObjs))
+		}
+	}
+}
+
+/*
```

Comment:
Why are these commented out? Are they no longer relevant or just the functionality is not yet implemented? Either remove them or provide a comment/TODO on how they will be resolved.

Reply:
this was left over as we had to refactor them. resolved now